ast/ast.go

```diff
@@ -64,6 +64,10 @@ type ExprNode interface {
 	SetValue(val interface{})
 	// GetValue gets value of the expression.
 	GetValue() interface{}
+	// SetDatum set datum to the expression.
+	SetDatum(datum *types.Datum)
```

Comment:
Use `types.Datum` instead of `*types.Datum` in `SetDatum`, this way can reduce memory allocation.